test/unit/deploy_spec_test.rb

```diff
@@ -359,5 +359,37 @@ class DeploySpecTest < ActiveSupport::TestCase
     test "#clear_working_directory? returns true by default" do
       assert_predicate @spec, :clear_working_directory?
     end
+
+    test '#npm? is false if there is no package.json' do
```

Comment:
Can we add a test to make sure that `npm` takes lower precedence than the other ones?